src/cpp/rtps/builtin/discovery/endpoint/EDP.cpp

```diff
@@ -504,6 +509,10 @@ bool EDP::unpairReaderProxy(
                 info.status = REMOVED_MATCHING;
                 info.remoteEndpointGuid = reader_guid;
                 (*wit)->getListener()->onWriterMatched((*wit),info);
+
+                GUID_t writer_guid = (*wit)->getGuid();
```

Comment:
This local variable is only used once, right? Then consider remove it.
If you prefer to keep it, store it as a const ref to avoid copies.